src/app/project/project-details-tab/project-details-tab.component.ts

```diff
@@ -156,40 +161,112 @@ export class ProjectDetailsTabComponent implements OnInit, AfterViewInit, OnDest
-        if (shapefile.getBounds().isValid() === true) {
-          this.map.fitBounds(shapefile.getBounds(), {padding: [50, 50]});
-          window.clearInterval(shapefileBounds);
-        }
-      }.bind(this), 500);
-      shapefile.addTo(this.map);
-    } else if (this.project) {
-      const markerIconYellow = L.icon({
-        iconUrl: 'assets/images/marker-icon-yellow.svg',
-        iconSize: [36, 36],
-        iconAnchor: [18, 36],
-        tooltipAnchor: [16, -28]
-      });
-
-      const title = `${this.project.name}\n`
-        + `${this.project.overlappingRegionalDistricts}\n`;
-      const marker = L.marker(L.latLng(this.project.centroid[1], this.project.centroid[0]), { title: title })
-        .setIcon(markerIconYellow);
-      this.map.addLayer(marker);
-    }
-    this.map.addLayer(this.appFG);
+		// Convert documents to shapefiles
+		this.shapefileStyle = { color: this.project.shapeFileColour || Constants.style.DEFAULT_SHAPEFILE_COLOUR }; // Colour for the shapefiles
+		this.convertedShapefiles = this.shapefiles[0]?.map(sf => {
+			if (sf?._id && sf?.documentFileName?.length > 0) {
+				const escapedName = encode(sf.documentFileName).replace(/\(/g, '%28').replace(/\)/g, '%29').replace(/\\/g, '_').replace(/\//g, '_').replace(/\%2F/g, '_');
+				const shapeurl = this.pathAPI + '/document/' + sf._id + '/fetch/' + escapedName;
+				return new L.Shapefile(shapeurl, { isArrayBuffer: false, style: this.shapefileStyle });
+			};
+		});
 
-    this.fixMap();
-  }
+    // draw project marker or shapefile(s)
+		let locationAdded = false;
+
+		// If there are converted shapefiles
+		if (this.convertedShapefiles) {
+			this.convertedShapefiles?.forEach((sf, index) => {
+				sf.on('data:loaded', () => {
+					const keys = Object.keys(sf._layers);
+					keys?.forEach(key => {
+						if (sf._layers[key]._bounds) {
+							this.calculateShapefileBounds(sf._layers[key]._bounds);
+						}
+					});
+
+					sf.addTo(this.appFG);
+					locationAdded = true;
+
+					// Last iteration only
+					if (index === this.convertedShapefiles?.length - 1) {
+						// Change map bounds based on values in bounds variable
+						if (this.bounds.southWest.lat && this.bounds.southWest.lng && this.bounds.northEast.lat && this.bounds.northEast.lng) {
+							this.map.fitBounds([[this.bounds.southWest.lat, this.bounds.southWest.lng], [this.bounds.northEast.lat, this.bounds.northEast.lng]], {padding: [50, 50]});
+						}
+						// If no shape files have been added, add a marker
+						if (!locationAdded && this.project) {
+							this.addMarker();
+						}
+						// Add the feature group to the map
+						this.map.addLayer(this.appFG);
+					}
+				})
+			});
+		} else {
+			// Otherwise skip the shapefiles and just add a marker
+			if (this.project) {
+				this.addMarker();
+				this.map.addLayer(this.appFG);
+				this.map.fitBounds(this.defaultBounds, {padding: [50, 50]});
+			}
+		}
+	}
+
+	/**
+	 * Caluclates the shapefile bounds.
```

Comment:
```suggestion
	 * Calculates the shapefile bounds.
```

Also wouldn't it be more accurate to say: "Takes the bounds of a shapefile and updates the overall map bounds to include them."

Reply:
Sure, I switched the comment to this.